src/Pos/Worker.hs

```diff
@@ -58,6 +60,12 @@ allWorkers NodeResources {..} = mconcatPair
     , wrap' "delegation" $ dlgWorkers
     , wrap' "slotting"   $ (properSlottingWorkers, mempty)
 
+    , case ncTopology ncNetworkConfig of
+        TopologyBehindNAT dnsDomains ->
+          wrap' "subscription" $ subscriptionWorkers ncNetworkConfig dnsDomains
+        _otherwise ->
+          mempty -- No subscription worker required
```

Comment:
So we already had `topologyNodeType`; @avieth added `topologySubscriberNodeType`, and I've just added `topologySubscriptionWorker`. We don't yet have `topologyRunKademlia` because this is currently based on the node resources rather than the topology; if we did try to base it on the topology, we'd then need to make some assumptions about the runtime values of those node resources (in particular, if `topologyRunKademlia` says `True`, then `nrKademlia` should be `Just` a value). Added a TODO.